app/pages/project/index.cjsx

```diff
@@ -82,6 +87,20 @@ ProjectPage = React.createClass
       .then (pages) =>
         @setState {pages}
 
+  getSelectedWorkflow: (project, preferences) ->
+    @setState selectedWorkflow: 'PENDING'
+
+    preferredWorkflowID = preferences?.preferences.selected_workflow ? project.configuration?.default_workflow
+    if preferredWorkflowID?
+      apiClient.type('workflows').get preferredWorkflowID
```

Comment:
sorry - just re-read..i really don't mind. I agree the code is explicit here without more knowledge of the workflow active filters.